ports/esp32/machine_adc_block.c

```diff
@@ -29,25 +29,21 @@
 
 #include "py/mphal.h"
 #include "adc.h"
-#include "driver/adc.h"
+#include "esp_adc/adc_oneshot.h"
 
 machine_adc_block_obj_t madcblock_obj[] = {
-    {{&machine_adc_block_type}, ADC_UNIT_1, SOC_ADC_RTC_MAX_BITWIDTH, -1, {0}},
+    {{&machine_adc_block_type}, ADC_UNIT_1, NULL},
```

Comment:
I think it's worth initialising the `bitwidth` field here, eg `.bitwidth = ...`.